internal/services/datafactory/data_factory_integration_runtime_azure_resource.go

```diff
@@ -272,11 +283,18 @@ func expandDataFactoryIntegrationRuntimeAzureComputeProperties(d *pluginsdk.Reso
 	location := azure.NormalizeLocation(d.Get("location").(string))
 	coreCount := int32(d.Get("core_count").(int))
 	timeToLiveMin := int32(d.Get("time_to_live_min").(int))
-	cleanup := true
-	// nolint staticcheck
-	if v, ok := d.GetOkExists("cleanup_enabled"); ok {
-		cleanup = v.(bool)
+
+	var cleanup bool
+	if features.FourPointOhBeta() {
+		cleanup = d.Get("cleanup_enabled").(bool)
+	} else {
+		cleanup = true
+		// nolint staticcheck
+		if v, ok := d.GetOkExists("cleanup_enabled"); ok {
+			cleanup = v.(bool)
+		}
 	}
```

Comment:
We were defaulting this to `true` in the expand before so we should still maintain that behaviour for now

```suggestion
	cleanup := true
	if features.FourPointOhBeta() {
		cleanup = d.Get("cleanup_enabled").(bool)
	} else {
		// nolint staticcheck
		if v, ok := d.GetOkExists("cleanup_enabled"); ok {
			cleanup = v.(bool)
		}
	}
```

Reply:
Updated